### ISSUE TITLE
Consider passing word embeddings through lossy PCA before sending to models (enhancement)

### ISSUE THREAD

**Issue by willsmithorg** [awslabs/autogluon#1494]
I've done some experiments with sentence vectors, typically we get ~500 vectors from the language model per input feature, which slows learning and increases memory usage.

**I tried passing them through `sklearn.decomposition.PCA(n_components=40)` or `=80`.** 

My motivation is that a sentence embedding contains so many things, conceptually things like whether it's past or future tense, is positive or negative, is singular or plural, involves animals or humans, etc. We often don't care about a lot of these, mostly our data is likely to be similar in many of these dimensions. So if we pass the 500+ vectors through PCA we compress down to only (orthogonal) components which vary across our data.   We can also, if necessary, measure how many components are needed to retains say 99% of the original variance.

**It worked well - I retained most of  the accuracy, with an expected reduction in memory and training time** I got most of the benefit of the 500+ sentence vectors using 40 or 80 components, giving a bit training and memory improvement. We should consider trying this in TextPredictor if we don't already, unless presets='best_quality'.

One thing I'm not clear on - how would we do this for inference?  I cheated and did the PCA on entire dataset. How do we instead train PCA only on the test data, and fit (apply) it only to the inference data?  I guess something like this: [https://stats.stackexchange.com/questions/144439/applying-pca-to-test-data-for-classification-purposes](https://stats.stackexchange.com/questions/144439/applying-pca-to-test-data-for-classification-purposes)

**Comment by willsmithorg** [comment]
As I now understand it, word embeddings are not used anywhere in AutoGluon, so the above suggestion can be closed:
- tabularpredictor doesn't use any word embeddings
- textpredictor creates neural network based transformer models directly from the text without going via word embeddings
-
